src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -1979,6 +1979,80 @@ void c_typecheck_baset::typecheck_typed_target_call(
   typecheck_side_effect_function_call(expr);
 }
 
+void c_typecheck_baset::typecheck_obeys_contract_call(
+  side_effect_expr_function_callt &expr)
+{
+  INVARIANT(
+    expr.function().id() == ID_symbol &&
+      to_symbol_expr(expr.function()).get_identifier() == CPROVER_PREFIX
+        "obeys_contract",
+    "expression must be a " CPROVER_PREFIX "obeys_contract function call");
+
+  if(expr.arguments().size() != 2)
+  {
+    throw invalid_source_file_exceptiont{
+      "expected 2 arguments for " CPROVER_PREFIX "obeys_contract, found " +
+        std::to_string(expr.arguments().size()),
+      expr.source_location()};
```

Comment:
There is a lot of handling of possible errors here, but seemingly none of these are being tested. I'd appreciate if at least some of these could be covered by regression tests.

Reply:
added some tests